File: frontend/src/services/brandService.js
import api from './api';

export const brandService = {
  getAllBrands: async () => {
    const response = await api.get('/brands');
    return response.data;
  },

  getBrandById: async (id) => {
    const response = await api.get(`/brands/${id}`);
    return response.data;
  },

  createBrand: async (brandData) => {
    const response = await api.post('/brands', brandData);
    return response.data;
  },

  updateBrand: async (id, brandData) => {
    const response = await api.put(`/brands/${id}`, brandData);
    return response.data;
  },

  deleteBrand: async (id) => {
    await api.delete(`/brands/${id}`);
  },
};

File: backend/Nexus-IMS/src/main/java/com/learn/nexus_ims/services/BrandService.java
package com.learn.nexus_ims.services;

import java.util.List;

import com.learn.nexus_ims.dtos.BrandDto;

public interface BrandService {
    BrandDto createBrand(BrandDto brandDto);
    BrandDto updateBrand(Long brandId, BrandDto brandDto);
    void deleteBrand(Long brandId);
    List<BrandDto> getAllBrands();
    BrandDto getBrandById(Long brandId);
}
